### PR TITLE
[Concept] Exercise - Loops

### PR DESCRIPTION
**Pull request by vaeng**
Add the interest-is-interesting exercise to explore loops.

### CODE REVIEW

File: exercises/concept/interest-is-interesting/.docs/introduction.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# Introduction
+
+## Loops
+
+Loops are used to repeatedly execute some logic.
+The two most common types are the `while loop` (indefinite looping) and the `for loop` (definite, or counted looping).
+There is also the `for each` loop, that will come up in a later concept.
+
```
Comment (meatball133) [inline exercises/concept/interest-is-interesting/.docs/introduction.md:8]: It is 2 new lines here I don't know if that is intentional or not.

File: exercises/concept/interest-is-interesting/.docs/introduction.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  some_statement;
+}
+```
+
+- The **init** component is some code that runs only once before the loop starts.
+- The **condition** component must be some expression that evaluates to a boolean and controls when the loop should stop.
+The code inside the loop will run as long as this condition evaluates to true.
+As soon as this expression evaluates to false, no more iterations of the loop will run.
+- The **expression** component is some code that will run at the end of each iteration.
+
```
Comment (meatball133) [inline exercises/concept/interest-is-interesting/.docs/introduction.md:27]: Same here

File: exercises/concept/interest-is-interesting/.docs/introduction.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+int equal_sum{0};
+for (int i{1}; i < 7; ++i) {
+  if  (n%2 == 1) {
+    continue;
+  }
+  equal_sum += i;
+}
+// equal_sum is now 12
+```
+> Note: it is usually easier to understand the logic of the loop, when the use of `break` and `continue` is minimized or entirely avoided.
```
Comment (meatball133) [inline exercises/concept/interest-is-interesting/.docs/introduction.md:96]: Perhaps use ~~~~exercism/note?

File: exercises/concept/interest-is-interesting/.docs/introduction.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+The header consists of 3 components separated by semicolons `;`:  init-statement, condition, and another expression.
+Each of these may be empty.
+
+```cpp
+for (init-statement; condition; expression) {
+  some_statement;
+}
+```
+
+- The **init** component is some code that runs only once before the loop starts.
```
Comment (meatball133) [inline exercises/concept/interest-is-interesting/.docs/introduction.md:22]: I think it would make sense to use the same naming as the example.

**Review by meatball133 — COMMENTED** [review]
Some small nits

**Review by BethanyG — APPROVED** [review]
Approving with the same comments that were left for the Arrays exercise.  Will re-review Arrays/Vectors when content is updated.

**Pull request merged**